partner_validate_vat/models/res_partner.py

```diff
@@ -0,0 +1,29 @@
+# coding: utf-8
+# Copyright 2017 Vauxoo (https://www.vauxoo.com) <info@vauxoo.com>
+# License LGPL-3.0 or later (http://www.gnu.org/licenses/lgpl).
+
+from openerp import models, api, fields, _
+from openerp.exceptions import ValidationError
+
+
+class ResPartner(models.Model):
+    _inherit = 'res.partner'
+
+    vat = fields.Char(copy=False)
+
+    @api.constrains('vat', 'country_id')
+    def _check_vat_uniqueness(self):
+        """ Check that the vat is unique in the level
+            where the partner in the tree
+        """
+
+        user_company = self.env.user.company_id
+
+        for partner in self:
+            current_vat = partner.commercial_partner_id.vat
+            if (user_company.country_id == partner.country_id and
```

Comment:
I try to validate the country of the partner in relation to the user's  country , because it may happen that he places the same vat in another partner located in another country without relation between them. For this case if it is accepted that the same vat is repeated for both partners.  This validation is important when placing the vat in the partner